pkg/daemon/daemon.go

```diff
@@ -1505,8 +1512,8 @@ func (dn *Daemon) triggerUpdateWithMachineConfig(currentConfig, desiredConfig *m
 	dn.stopConfigDriftMonitor()
 
 	// Hack in our layered node workflow for pools labeled as "layered"
-	if dn.isLayeredNode() {
-		return dn.experimentalUpdateLayeredConfig()
+	if coreOSDaemon := dn.shouldUpdateLayered(); coreOSDaemon != nil {
```

Comment:
I feel like you're kind of getting at a thing with CoreOSDaemon where we have different "types"/interfaces of daemons that determine their "features" that we could check.